From c1a87e55e1071f3a6507d638e184ec6f71f5e4e4 Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@st.com>
Date: Wed, 14 Sep 2022 17:12:59 +0200
Subject: [PATCH] net: phy: Add condition PHY_NOLINK to delete warning in
 mdio_bus_phy_resume()

NOT_UPSTREAMABLE only avoid verbose message for valid purpose.
And found why in some cases this warning is displayed
(almost 80% of cases)

Signed-off-by: Christophe Roullier <christophe.roullier@st.com>
Change-Id: I3c6865ebddd2d6954832c352d538e9ea94ad35a0
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/266290
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/396339
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Christophe ROULLIER <christophe.roullier@foss.st.com>
Domain-Review: Christophe ROULLIER <christophe.roullier@foss.st.com>
Tested-by: Christophe ROULLIER <christophe.roullier@foss.st.com>
---
 drivers/net/phy/phy_device.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/phy/phy_device.c b/drivers/net/phy/phy_device.c
index c895cd178e6a..79e014493ef2 100644
--- a/drivers/net/phy/phy_device.c
+++ b/drivers/net/phy/phy_device.c
@@ -358,7 +358,7 @@ static __maybe_unused int mdio_bus_phy_resume(struct device *dev)
 	 * MAC managed PM, but we are not.
 	 */
 	WARN_ON(phydev->state != PHY_HALTED && phydev->state != PHY_READY &&
-		phydev->state != PHY_UP);
+		phydev->state != PHY_UP && phydev->state != PHY_NOLINK);
 
 	ret = phy_init_hw(phydev);
 	if (ret < 0)
-- 
2.39.5

